util/src/main/java/org/cryptomator/util/crypto/BiometricAuthCryptor.java

```diff
@@ -17,21 +17,29 @@ public class BiometricAuthCryptor {
 	private static final String BIOMETRIC_AUTH_KEY_ALIAS = "fingerprintCryptoKeyAccessToken";
 
 	private final Cipher cipher;
+	private final CryptoMode cryptoMode;
 
-	private BiometricAuthCryptor(Context context) throws UnrecoverableStorageKeyException {
+	private BiometricAuthCryptor(Context context, CryptoMode cryptoMode) throws UnrecoverableStorageKeyException {
+		String suffixedAlias = getSuffixedAlias(cryptoMode);
 		KeyStore keyStore = KeyStoreBuilder.defaultKeyStore() //
-				.withKey(BIOMETRIC_AUTH_KEY_ALIAS, true, context) //
+				.withKey(suffixedAlias, true, cryptoMode, context) //
 				.build();
-		this.cipher = CryptoOperationsFactory.cryptoOperations().cryptor(keyStore, BIOMETRIC_AUTH_KEY_ALIAS);
+		this.cryptoMode = cryptoMode;
+		this.cipher = CryptoOperationsFactory.cryptoOperations(cryptoMode).cryptor(keyStore, suffixedAlias);
 	}
 
-	public static BiometricAuthCryptor getInstance(Context context) throws UnrecoverableStorageKeyException {
-		return new BiometricAuthCryptor(context);
+	private static String getSuffixedAlias(CryptoMode cryptoMode) {
+		// CBC does not have an alias due to legacy reasons
+		return cryptoMode == CryptoMode.GCM ? BIOMETRIC_AUTH_KEY_ALIAS + "_GCM" : BIOMETRIC_AUTH_KEY_ALIAS;
```

Comment:
I'd suggest also using a switch-statement here as you did in `BiometricAuthenticationMigration:`
https://github.com/cryptomator/android/blob/7a7f477822aca0174178c3c2b98641a56ec14b54/presentation/src/main/java/org/cryptomator/presentation/util/BiometricAuthenticationMigration.kt#L104-L108

Using an if-else might cause us to forget updating this section of the code later on.

Reply:
Fixed in https://github.com/cryptomator/android/commit/9615090966def68d10b86241c37ddace9946c0ef